addons/mail/models/mail_link_preview.py

```diff
@@ -49,42 +46,39 @@ def _create_from_message_and_notify(self, message, request_url=None):
         for url in urls:
             if ignore_pattern and ignore_pattern.match(url):
                 continue
+            existing_link_preview = link_previews.search([("source_url", "=", url)])
+            if existing_link_preview and not url in link_previews_by_url:
+                self.env["mail.link.preview.message"].create({
+                    "message_id": message.id,
+                    "link_preview_id": existing_link_preview[0].id
+                })
+                link_previews += existing_link_preview
+                continue
```

Comment:
Could be great to accumulate based on urls and search once, create once. Try to write multi-enabled code.